src/common/ScreenExporter.hpp

```diff
@@ -0,0 +1,58 @@
+/* *****************************************************************************
+ * A.L.E (Arcade Learning Environment)
+ * Copyright (c) 2009-2013 by Yavar Naddaf, Joel Veness, Marc G. Bellemare and 
+ *   the Reinforcement Learning and Artificial Intelligence Laboratory
+ * Released under the GNU General Public License; see License.txt for details. 
+ *
+ * Based on: Stella  --  "An Atari 2600 VCS Emulator"
+ * Copyright (c) 1995-2007 by Bradford W. Mott and the Stella team
+ *
+ * *****************************************************************************
+ *  ScreenExporter.hpp 
+ *
+ *  A class for exporting Atari 2600 frames as PNGs.
+ *
+ **************************************************************************** */
+
+#ifndef __SCREEN_EXPORTER_HPP__
+#define __SCREEN_EXPORTER_HPP__ 
+
+#include <string>
+#include "display_screen.h"
+#include "ale_screen.hpp"
+
+class ScreenExporter {
+
+    public:
+
+        /** Creates a new ScreenExporter which can be used to save screens using save(filename). */ 
+        ScreenExporter(ColourPalette &palette);
```

Comment:
After further thought I think it is fine as it is (taking a ColorPalette as an argument). I think we should hide the Screen Exporter behind the Stella Environment since Stella Environment is the one to handle the command line flag for recording games.